p/api/fever.php

```diff
@@ -0,0 +1,894 @@
+		$res = $doc->saveHTML();
+		return $res;
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function getItems()
+	{
+		$feed_ids = [];
+		$entry_ids = [];
+		$max_id = null;
+		$since_id = null;
+
+		if (isset($_REQUEST["feed_ids"]) || isset($_REQUEST["group_ids"])) {
+			if (isset($_REQUEST["feed_ids"])) {
+				$feed_ids = explode(",", $_REQUEST["feed_ids"]);
+			}
+
+			$dao = $this->getDaoForCategories();
+			if (isset($_REQUEST["group_ids"])) {
+				$group_ids = explode(",", $_REQUEST["group_ids"]);
+				foreach ($group_ids as $id) {
+					/** @var FreshRSS_Category $category */
+					$category = $dao->searchById($id);
+					/** @var FreshRSS_Feed $feed */
+					foreach ($category->feeds() as $feed) {
+						$feeds[] = $feed->id();
+					}
+				}
+
+				$feed_ids = array_unique($feeds);
+			}
+		}
+
+		if (isset($_REQUEST["max_id"])) {
+			// use the max_id argument to request the previous $item_limit items
+			if (is_numeric($_REQUEST["max_id"])) {
+				$max = ($_REQUEST["max_id"] > 0) ? intval($_REQUEST["max_id"]) : 0;
+				if ($max) {
+					$max_id = $max;
+				}
+			}
+		} else if (isset($_REQUEST["with_ids"])) {
+			$entry_ids = explode(",", $_REQUEST["with_ids"]);
+		} else {
+			// use the since_id argument to request the next $item_limit items
+			$since_id = isset($_REQUEST["since_id"]) && is_numeric($_REQUEST["since_id"]) ? intval($_REQUEST["since_id"]) : 0;
+		}
+
+		$items = [];
+
+		$dao = $this->getDaoForEntries();
+		$entries = $dao->findEntries($feed_ids, $entry_ids, $max_id, $since_id);
+
+		foreach($entries as $entry) {
+			$items[] = [
+				"id" => $entry->id(),
+				"feed_id" => $entry->feed(false),
+				"title" => $entry->title(),
+				"author" => $entry->author(),
+				"html" => $this->sanitizeContent($entry->content()),
+				"url" => $entry->link(),
+				"is_saved" => $entry->isFavorite() ? 1 : 0,
+				"is_read" => $entry->isRead() ? 1 : 0,
+				"created_on_time" => $entry->date(true)
+			];
+		}
+
+		return $items;
+	}
+
+	/**
+	 * TODO replace by a dynamic fetch for id <= $before timestamp
+	 *
+	 * @param int $beforeTimestamp
+	 * @return int
+	 */
+	protected function convertBeforeToId($beforeTimestamp)
+	{
+		// if before is zero, set it to now so feeds all items are read from before this point in time
+		if ($beforeTimestamp == 0) {
+			$before = time();
+		}
+		$before = PHP_INT_MAX;
+
+		return $before;
+	}
+
+	protected function setFeedAsRead($id, $before)
+	{
+		$before = $this->convertBeforeToId($before);
+		$dao = $this->getDaoForEntries();
+		$dao->markReadFeed($id, $before);
+	}
+
+	protected function setGroupAsRead($id, $before)
+	{
+		$before = $this->convertBeforeToId($before);
+		$dao = $this->getDaoForEntries();
+		$dao->markReadCat($id, $before);
+	}
+}
+
+// ================================================================================================
+// Start the Fever API handling
+$handler = createFeverApiInstance();
+
+if ($handler->isXmlRequested()) {
+	header("Content-Type: text/xml");
```

Comment:
The following would probably be better: `Content-Type: application/xml; charset=UTF-8`